lang/funcs/filesystem.go

```diff
@@ -237,6 +237,21 @@ var DirnameFunc = function.New(&function.Spec{
 	},
 })
 
+// AbsPathFunc constructs a function that converts a filesystem path to an absolute path
+var AbsPathFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "path",
+			Type: cty.String,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		absPath, err := filepath.Abs(args[0].AsString())
```

Comment:
Another filepath-related change in 0.12 is that `path.module`, `path.cwd`, and `path.root` all always use forward slashes, regardless of host platform, so that common patterns like `${path.module}/foo.txt` can work as expected on Windows.

For consistency, I think we'd better pass the result of this through `filepath.ToSlash` to ensure we don't suddenly switch to backslashes when `path.root` is replaced with `abspath(path.root)`.